tutorials/rabi-oscillations.md

```diff
@@ -0,0 +1,187 @@
+---
+jupyter:
+  jupytext:
+    text_representation:
+      extension: .md
+      format_name: markdown
+      format_version: '1.3'
+      jupytext_version: 1.13.8
+  kernelspec:
+    display_name: Python 3 (ipykernel)
+    language: python
+    name: python3
+---
+
+# QuTiP example: Vacuum Rabi oscillations in the Jaynes-Cummings model
+
+Authors: J.R. Johansson and P.D. Nation
+
+Slight modifications: C. Staufenbiel (2022)
+
+This notebook demonstrates how to simulate the quantum vacuum rabi 
+oscillations in the Jaynes-Cumming model, using the Master Equation Solver 
+`qutip.mesolve`. We also consider the dissipative version of the 
+Jaynes-Cumming model, i.e., the cavity and the atom are coupled to an 
+environment.
+
+
+For more information on the theory behind the Master Equation Solver see [the documentation](https://qutip.org/docs/latest/guide/dynamics/dynamics-master.html#non-unitary-evolution).
+
+
+### Package import
+
+```python
+%matplotlib inline
+import matplotlib.pyplot as plt
+import numpy as np
+from qutip import tensor, basis, destroy, qeye, mesolve
+```
+
+# Introduction
+
+The Jaynes-Cumming model is the simplest possible model of quantum mechanical light-matter interaction, describing a single two-level atom interacting with a single electromagnetic cavity mode. The Hamiltonian for this system is (in dipole interaction form)
+
+$H = \hbar \omega_c a^\dagger a + \frac{1}{2}\hbar\omega_a\sigma_z + \hbar g(a^\dagger + a)(\sigma_- + \sigma_+)$
+
+or with the rotating-wave approximation
+
+$H_{\rm RWA} = \hbar \omega_c a^\dagger a + \frac{1}{2}\hbar\omega_a\sigma_z + \hbar g(a^\dagger\sigma_- + a\sigma_+)$
+
+where $\omega_c$ and $\omega_a$ are the frequencies of the cavity and atom, respectively, and $g$ is the interaction strength.
+
+In this example we also consider the coupling of the Jaynes-Cummings model to an external environment, i.e., we need to solve the system using the Master Equation Solver `qutip.mesolve`. The coupling to the environment is described by the collapse operators (as described in [the docs](https://qutip.org/docs/latest/guide/dynamics/dynamics-master.html#non-unitary-evolution)). Here, we consider two collapse operators for the cavity $C_1, C_2$, describing creation and annihilation of photons, and one collapse operator for the atom $C_3$.
+
+$C_1 = \sqrt{\kappa (1+T)} a$
+
+$C_2 = \sqrt{\kappa T} a^\dagger$
+
+$C_3 = \sqrt{\gamma} \sigma_-$
+
+where $T$ is the temperature of the environment (in units of the frequency). By setting $T=0$ we remove the creation of photons and only consider the annihilation of photons.
+
```

Comment:
I think it's a bit clearer and more general to define the rates in terms of thermal occupation n_th here. can also use the utility function https://github.com/qutip/qutip/blob/d22d0abe7af7ff049bcb3a0fb016af91b8a3febf/qutip/utilities.py#L10
to define it in terms of temperature

Reply:
I guess, the definition used in the code is the average photon number. I just renamed it in the markdown description. What do you think?